File: 01259.py
while 1:
    s = input()
    if s == '0': break

    isPal = True
    for i in range((len(s) - 1) // 2 + 1):
        if s[i] != s[-i-1]:
            isPal = False
            break

    if isPal: print("yes")
    else: print("no")

File: 01259.cpp
#include <iostream>
using namespace std;

int main() {
	while (1) {
		string s; cin >> s;
		if (s == "0") break;

		bool isPal = true;
		for (int i = 0; i <= (s.length() - 1) / 2; i++) {
			if (s[i] != s[s.length() - 1 - i]) {
				isPal = false;
				break;
			}
		}

		if (isPal) cout << "yes\n";
		else cout << "no\n";
	}
}
